homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -74,8 +77,16 @@ def nth_departure_time(self, i: int) -> datetime | None:
         return None
 
     async def _async_update_data(self) -> list[DataConnection]:
+        return await self.fetch_connections(limit=SENSOR_CONNECTIONS_COUNT)
+
+    async def fetch_connections(self, limit: int) -> list[DataConnection]:
+        """Fetch connections using the opendata api."""
+        self._opendata.limit = limit
         try:
             await self._opendata.async_get_data()
+        except OpendataTransportConnectionError:
+            _LOGGER.warning("Connection to transport.opendata.ch cannot be established")
+            raise
```

Comment:
Should raise `UpdateFailed` I think and we can remove the logging. Or perhaps better to use a try...except block in `async_update_data` instead as this method is also used by the new service.